### PR TITLE
header implemented with tsx, folder structure and styled components

### PR DESCRIPTION
**Pull request by buddharajuprithvi**
header implemented with

- tsx
- folder structure
- styled components

### CODE REVIEW

File: src/components/Header/__snapshots__/Header.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,115 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (balavishnuvj) [inline src/components/Header/__snapshots__/Header.test.js.snap]: the snapshot doesn't seem like a proper representation of the header components. Will ignore this for now and will update when we setup the test properly.

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  HeaderContainer,
+} from './header.styled';
+import IogoSmall from './logo-small.png';
+
+interface IProps {
+  testingProps?: string;
+}
+
+function Header(props: IProps) {
+  return (
```
Comment (balavishnuvj) [inline src/components/Header/Header.tsx:13]: destructure all the props at top, here `const {testingProps} = props`

File: src/components/Header/Header.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import Header from './Header';
+import createApolloClient from '../../utils/shared/apollo';
+
+describe('Header', () => {
+  test('renders children correctly', () => {
+    const client = createApolloClient();
+    const wrapper = renderer
+      .create(
+        <Header
+          context={{
```
Comment (balavishnuvj) [inline src/components/Header/Header.test.js]: context isn't an appropriate prop for the header. But will fix this later.

File: src/components/Layout/Layout.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ class Layout extends React.Component {
     return (
       <ThemeProvider theme={theme}>
         <div>
-          <Header />
+          <Header testingProps="Sri Kanya" />
```
Comment (balavishnuvj) [inline src/components/Layout/Layout.js]: remove this change.

File: src/d.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+declare module '*.png' {
```
Comment (balavishnuvj) [inline src/d.ts]: relocate this file to `src/@types/global.d.ts`

File: src/components/Header/header.styled.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
```
Comment (balavishnuvj) [inline src/components/Header/header.styled.ts]: rename this file to `Header.styled.ts`, capital *H* for consistency.

**Review by balavishnuvj — CHANGES_REQUESTED** [review]

File: package.json [hunk 7]
```diff
@@ -24,7 +24,6 @@
     "@types/node-fetch": "^2.1.4",
     "@types/react": "^16.7.20",
     "@types/react-dom": "^16.0.11",
-    "@types/react-test-renderer": "^16.0.3",
```
Comment (balavishnuvj) [inline package.json]: why these changes?

File: src/components/Header/header.styled.ts (hunk 6 above)
Comment (balavishnuvj) [reply src/components/Header/header.styled.ts]: file now changed

File: src/components/Layout/__tests__/__snapshots__/Layout.test.tsx.snap [hunk 8]
```diff
@@ -3,82 +3,17 @@
 exports[`renders children correctly 1`] = `
 <div>
   <div
-    class="root"
+    class="sc-bdVaJa kttCLS"
```
Comment (balavishnuvj) [inline src/components/Layout/__tests__/__snapshots__/Layout.test.tsx.snap:6]: how come this changed when u didn't change the layout?

File: yarn.lock [hunk 9]
```diff
@@ -742,13 +742,6 @@
     pirates "^4.0.0"
     source-map-support "^0.5.9"
 
-"@babel/runtime@^7.1.5":
```
Comment (balavishnuvj) [inline yarn.lock]: why these changes?

**Review by balavishnuvj — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (buddharajuprithvi) [reply package.json]: that came when I took a pull.
I dint remove it.

File: src/components/Layout/__tests__/__snapshots__/Layout.test.tsx.snap (hunk 8 above)
Comment (buddharajuprithvi) [reply src/components/Layout/__tests__/__snapshots__/Layout.test.tsx.snap:6]: when I did yarn test -u
maybe then this much be updated

**Review by balavishnuvj — APPROVED** [review]

**Pull request merged**